cordapp-contracts-states/src/main/kotlin/com/negotiation/StatesAndContracts.kt

```diff
@@ -31,10 +31,32 @@ open class ProposalAndTradeContract : Contract {
                 "There is no timestamp" using (tx.timeWindow == null)
             }
             is Commands.Accept -> requireThat {
-                // TODO: Add contractual logic.
+                val input = tx.inputsOfType<ProposalState>().single()
+                val output = tx.outputsOfType<TradeState>().single()
+                "There is exactly one output" using (tx.outputStates.size == 1)
+                "There is exactly one input" using (tx.inputStates.size == 1)
+
+                "The amount should be same after accepting the proposal" using (output.amount == input.amount)
+
+                "The proposer is a required signer" using (cmd.signers.contains(input.proposer.owningKey))
+                "The proposee is a required signer" using (cmd.signers.contains(input.proposee.owningKey))
+
+                "The buyer is a required signer" using (cmd.signers.contains(output.buyer.owningKey))
```

Comment:
If we check that the buyer and seller are unchanged (my suggestion above), then there is no need to check that the buyer/seller are required signers IMO, since we know that during issuance the buyer/seller had to be the proposer/proposee.